File: app/src/main/java/de/bitb/astroskop/ui/base/MVPActivity.java
package de.bitb.astroskop.ui.base;

import android.os.Bundle;
import android.support.annotation.Nullable;

import de.bitb.astroskop.AstroApplication;
import lombok.Getter;

public abstract class MVPActivity<T extends IBaseView, P extends BasePresenter<T>> extends BaseActivity {

    @Getter
    private P presenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        presenter = createPresenter((AstroApplication) getApplication());
    }

    protected abstract P createPresenter(AstroApplication application);
}

File: app/src/main/java/de/bitb/astroskop/ui/base/MVPActivity.kt
package de.bitb.astroskop.ui.base

import android.os.Bundle

import de.bitb.astroskop.BaseApplication
import lombok.Getter

abstract class MVPActivity<T : IBaseView, P : BasePresenter<T>> : BaseActivity() {

    @Getter
    var presenter: P? = null
        private set

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        presenter = createPresenter(application as BaseApplication)
    }

    protected abstract fun createPresenter(application: BaseApplication): P
}
